### DISCUSSION TITLE
Url rewrite from commandbox to apache tomcat error

### DISCUSSION THREAD

**Discussion by MvdO79** [wheels-dev/wheels#1394] [Q&A]
Moving a project to production (from commandbox rewrites to apache tomcat) with rewrite enabled I have this .htaccess in the root:

`<IfModule mod_rewrite.c>
Options +FollowSymLinks
RewriteEngine On
RewriteCond %{REQUEST_URI} ^.*/index.cfm/(.*)$ [NC]
RewriteRule ^.*/index.cfm/(.*)$ ./rewrite.cfm/$1 [NS,L]
RewriteCond %{REQUEST_URI} !^.*/(flex2gateway|jrunscripts|cfide|cf_scripts|cfformgateway|cffileservlet|lucee|files|images|javascripts|miscellaneous|stylesheets|wheels/public/assets|robots.txt|favicon.ico|sitemap.xml|rewrite.cfm)($|/.*$) [NC]
RewriteRule ^(.*)$ ./rewrite.cfm/$1 [NS,L]
</IfModule>`

The index.cfm still exists in the path and when trying a route without the index I get an error:

Type: Status Report
Message: The requested resource [/rewrite.cfm/user/marco/9f1c2] is not available
Description: The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.

So it's on shared hosting which might have an impact on how to setup the .htaccess, I am already asking support by them to check stuff on their end, but any thoughts? Thanks!

**Comment by dbelanger** [comment]
I've just come across this same problem when migrating from IIS to Apache (Wheels 2.5.1) - Either I get a blank page or I get a "not available" - When I try to hit the .root page, I simply get /rewrite/? in the browser

I see question was closed - Did you find a solution? I spend the later half of Friday trying to work this out.

**Reply by neokoenig** [reply]
If you're on lucee,have you added /rewrite.com/* to the URL rewriting page in web.xml (or server.xml, can't remember which)

**Reply by dbelanger** [reply]
And's with one line, Tom has solved the problem.  

So, that needs to be added to the Apache instructions too as simply enabling the .htaccess no longer works.  I added the change Tom suggested to Lucee's global `/opt/lucee/tomcat/conf/web.xml` in the CFMLServlet section.

Also, if we look at the examples from urlrewrite and web.config in the docs, neither have the first two lines the .htaccess example has.  I've removed the lines
```
RewriteCond %{REQUEST_URI} ^.*/index.cfm/(.*)$ [NC]
RewriteRule ^.*/index.cfm/(.*)$ ./rewrite.cfm/$1 [NS,L]
```
from .htaccess and the site continues to work.

**Reply by MvdO79** [reply]
I am still trying to solve the problem, I closed it as I thought it was solved... Gonna try Tom's suggestion. I can update the docs to reflect above insights, will do!

**Reply by MvdO79** [reply]
For setting up apache.

On my shared hosting I can access "/WEB-INF" which has also a folder "lucee" in it.

Is it possible, and where, should I put the web.xml, urlrewrite.xml?
And where should I put the .htaccess file?

Or can / should I add the rewriting "/rewrite.com/*" in lucee/lucee-web.xml.cfm?

Or... is it really up to the hosting provider to put the rewriting in the global web.xml?

**Reply by dbelanger** [reply]
For shared environments, you can only do it locally and it should still be something like `WEB-INF/lucee/tomcat/conf/web.xml`.  I don't have my shared environment anymore so I can't verify but I'm pretty sure that's how Lucee v5 handled multi-site.
For Apache, you don't need urlrewrite.xml, just that web.xml change and your **.htaccess**
The instructions for Apache are good, you just need to add a few lines about adding rewrite.cfm to the web.xml for Lucee servers.

**Reply by MvdO79** [reply]
Thanks @dbelanger , so just mimic the global path in the local folder, I will try that.
